framework/src/controller/bus.js

```diff
@@ -178,17 +183,46 @@ class Bus extends EventEmitter2 {
 			throw new Error(`Event ${eventName} is not registered to bus.`);
 		}
 
-		super.emit(eventName, eventValue); // Communicate throw event emitter
-		this.pubSocket.emit(eventName, eventValue); // Communicate throw unix socket
+		// Communicate through event emitter
+		super.emit(eventName, eventValue);
+
+		// Communicate through unix socket
+		if (this.config.ipc.enabled) {
+			this.pubSocket.emit(eventName, eventValue);
+		}
 	}
 
 	subscribe(eventName, cb) {
 		if (!this.getEvents().includes(eventName)) {
-			throw new Error(`Event ${eventName} is not registered to bus.`);
+			this.logger.info(
+				`Event ${eventName} was subscribed but not registered to the bus yet.`
+			);
+		}
+
+		// Communicate through event emitter
+		super.on(eventName, cb);
+
+		// Communicate through unix socket
+		if (this.config.ipc.enabled) {
+			this.subSocket.on(eventName, cb);
 		}
+	}
 
-		super.on(eventName, cb); // Communicate throw event emitter
-		this.subSocket.on(eventName, cb); // Communicate throw unix socket
+	once(eventName, cb) {
+		if (!this.getEvents().includes(eventName)) {
+			this.logger.info(
+				`Event ${eventName} was subscribed but not registered to the bus yet.`
+			);
+		}
+
+		// Communicate through event emitter
+		super.once(eventName, cb);
```

Comment:
`this.once`?

Reply:
Here we can't use `this` because `this.once` is its own method and we get `Maximum call stack size exceeded`

Reply:
ah my bad, this will be recursive... =p